### PR TITLE
Use `[SecureContext]` instead of manually throwing an exception

### PR DESCRIPTION
**Pull request by romandev**

### CODE REVIEW

**Comment by kenchris** [comment]
@alexshalamov are you fine with this? I assume this is the same you do in generic sensors?

File: index.html [hunk 1]
```diff
@@ -1462,6 +1463,9 @@ <h2>The <dfn>NFCRecordType</dfn> enum</h2>
     cancelable Promises</a> used with <code>push()</code> and
     <code>watch()</code>, respectively.
   </p>
+  <p class="note">
```
Comment (alexshalamov) [inline index.html]: I'm not sure whether this extra information adds something, localhost considered to be secure, that is sufficient for development needs, thus UA don't need to handle special case.

File: index.html [hunk 2]
```diff
@@ -1765,18 +1769,6 @@ <h2>The <dfn>NFCWatchMode</dfn> enum</h2>
             Return a new <a><code>Promise</code></a> <var>promise</var>, and
             then continue running this algorithm <a>in parallel</a>.
           </li>
-          <li>
-            If any exception occurs while running these steps, reject
```
Comment (alexshalamov) [inline index.html]: This block is not related to secure context, we can keep it as it is.

File: index.html [hunk 3]
```diff
@@ -2656,18 +2640,6 @@ <h2>The <dfn>NFCWatchMode</dfn> enum</h2>
             Return <var>promise</var> and continue the following steps
             <a>in parallel</a>.
           </li>
-          <li>
-            If any exception occurs while running these steps, reject
```
Comment (alexshalamov) [inline index.html]: Same as above, let's remove custom handling of secure context and remove only following blocks:

```
-          <li>
-            If the <a>incumbent settings object</a> is not a
-            <a>secure context</a>, reject <var>promise</var> with
-            <code>"<a>SecurityError</a>"</code> and abort these steps.
-            <div class="note">
-              Browsers may ignore this rule for development purposes only.
-            </div>
-          </li>
```

**Review by alexshalamov — CHANGES_REQUESTED** [review]

**Comment by romandev** [comment]
@alexshalamov Thank you for review. I addressed your comments.

**Review by jyasskin — APPROVED** [review]

**Review by alexshalamov — APPROVED** [review]

**Comment by alexshalamov** [comment]
@romandev Thanks!

**Pull request merged**